src/reducers/MapReducer.js

```diff
@@ -47,10 +54,71 @@ const INITIAL_STATE = {
     },
     mapLoaded: false,
     viewStyle: map.SearchOptions.MAP,
-    sites: []
+    sites: [],
+    displaySites: [],
+    filterCriteriaKeys: {accessibility: [], facilities: [], price: []}
+};
+
+const updateFilterKeys = ({filterCriteriaKeys}, filterKey) => {
+    let filterCriteriaSubKey = null;
+
+    if (_.includes(_.keys(accessibility_options), filterKey)) {
+        filterCriteriaSubKey = 'accessibility';
+    } else if (_.includes(_.keys(facilities_options), filterKey)) {
+        filterCriteriaSubKey = 'facilities';
+    } else if (_.includes(_.keys(price_options), filterKey)) {
+        filterCriteriaSubKey = 'price';
+    }
```

Comment:
Since `filterCriteriaKeys` is nested, we need to know which sub key to look inside of to see if we already have the given key. Above we check to find out which sub key it should be.